Misc

Mingling94 opened the pull request:
- Changed travis line for virtual display
- Load confi.json via require instead of fs
- bumped electron version
- Electron's module structure changed, updated require statements
- Misc other fixes detailed in commit headlines

DavidVorick commented on lib/daemon.js:
```diff
@@ -46,13 +39,13 @@ module.exports = {
     stop: function() {
         if (!daemonProcess) {
-            console.error("Daemon process not running!");
+            console.error("Daemon.js> process not running!");
```
daemon.js>

Mingling94 committed:
Changed virtual display that travis loads

Got the following email from the writer of [this blog
post](http://rhysd.hatenablog.com/entry/2015/08/07/181418)
> Hi,
>
> I'm glad to hear that!
>
> After that blog post, I found that below is better configuration for Travis CI.
>
>   - sh -e /etc/init.d/xvfb start
>
> is better than
>
>   - Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
>
> (if you don't need to specify screen size.)
>
> Detailed&official information is here.
>
> http://docs.travis-ci.com/user/gui-and-headless-browsers/
>
> Good luck!
> …
>
>
>
> ----- Original Message -----
> >From: Ming Luo <mcluo1994@gmail.com>
> >To: lin90162@yahoo.co.jp
> >Date: 2015/11/12, Thu 09:17
> >Subject: Thank You for your Blog Post
> >
> >
> >Hello,
> >
> >Your blog post was the ONE source on the internet that helped me understand why my mocha tests weren't working on TravisCI for my electron project.
> >
> >Thank you for publishing it!
> >--
> >
> >Sincerely and cheerily,
> >Ming Luo
> >Rensselaer Polytechnic Institute '16
> >B.S. Computer Science
> >Cell: (434) 420-2952
> >mcluo1994@gmail.com
> >
> >

Mingling94 committed:
Can load json files with require

Mingling94 committed:
Bumped electron-prebuilt to 0.35.0

Mingling94 committed:
Not all the npm release commands were zipping up

Mingling94 committed:
Readme fix link

Mingling94 committed:
Ignore errors.log file

Mingling94 committed:
Wasn't interpreting missing siad error correctly

Mingling94 committed:
Updated electron module requirements

Mingling94 commented:
@lukechampine @DavidVorick boop

Mingling94 committed:
Have travis lint when testing

DavidVorick commented on package.json:
```diff
@@ -24,20 +24,20 @@
+    "release-lin32": "npm run release-args -- --platform=linux --arch=ia32 && ./.finalizeRelease.sh",
+    "release-win64": "npm run release-args -- --platform=win32 --arch=x64 && ./.finalizeRelease.sh",
+    "release-win32": "npm run release-args -- --platform=win32 --arch=ia32 && ./.finalizeRelease.sh",
+    "release-mac": "npm run release-args -- --platform=darwin --arch=x64 && ./.finalizeRelease.sh"
```
can you explain this?

Mingling94 replied:
takes out `default_app` and zips up the release for each of the individual release commands.

Mingling94 committed:
Simplified some config keys

Mingling94 committed:
Made size/pos saving simpler

Mingling94 committed:
Fixed some things according to changes to electron

0.35.0 for electron introduced different behavior for IPCRenderer that
was unaccounted for in plugins

the contextMenu and appMenu also weren't up to date

Mingling94 committed:
Test bounds

Mingling94 commented:
can this be merged now? @lukechampine @DavidVorick
